lib/OpenFeature/DevCycleProvider.php

```diff
@@ -0,0 +1,138 @@
+     * @return void
+     */
+    public function setLogger(LoggerInterface $logger): void
+    {
+        // TODO: Implement setLogger() method.
+    }
+
+    /**
+     * @return Metadata
+     */
+    public function getMetadata(): IMetadata
+    {
+        return $this->metadata;
+    }
+
+    /**
+     * @param string $flagKey
+     * @param bool $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveBooleanValue(string $flagKey, bool $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        if ($context == null) {
+            throw new \InvalidArgumentException("Context cannot be null");
+        }
+        $user = DevCycleUser::FromEvaluationContext($context);
+
+        return $this->apiClient->variable($user, $flagKey, $defaultValue)->asResolutionDetails();
+    }
+
+    /**
+     * @param string $flagKey
+     * @param string $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveStringValue(string $flagKey, string $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        if ($context == null) {
+            throw new \InvalidArgumentException("Context cannot be null");
+        }
+        $user = DevCycleUser::FromEvaluationContext($context);
+
+        return $this->apiClient->variable($user, $flagKey, $defaultValue)->asResolutionDetails();
+    }
+
+    /**
+     * @param string $flagKey
+     * @param int $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveIntegerValue(string $flagKey, int $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        if ($context == null) {
+            throw new \InvalidArgumentException("Context cannot be null");
+        }
+        $user = DevCycleUser::FromEvaluationContext($context);
+
+        return $this->apiClient->variable($user, $flagKey, $defaultValue)->asResolutionDetails();
+    }
+
+    /**
+     * @param string $flagKey
+     * @param float $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveFloatValue(string $flagKey, float $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        if ($context == null) {
+            throw new \InvalidArgumentException("Context cannot be null");
+        }
+        $user = DevCycleUser::FromEvaluationContext($context);
+
+        return $this->apiClient->variable($user, $flagKey, $defaultValue)->asResolutionDetails();
+    }
+
+    /**
+     * @param string $flagKey
+     * @param array $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveObjectValue(string $flagKey, array $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
```

Comment:
This will need a bunch of tests to check for all the different JSON restrictions: https://docs.devcycle.com/sdk/server-side-sdks/node/node-openfeature#json-flag-limitations

Reply:
This one is a bit tricky - but yes; the handling in PHP is a bit cleaner than other languages thanks to the array concept being basically a dict.